### DISCUSSION TITLE
Weather API - not working for me.

### DISCUSSION THREAD

**Discussion by TEBrewMeister** [jgyates/genmon#1145] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

Genmon
Version V1.19.03

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have submitted logs

### What is/are your question(s)?

I'm new to RPIs - just got am Zero 2 up and running on my 22KW Honeywell.
Turned on the weather option.
Pasted in my API Key and Location code.
Tried it with minimal display on and off.
I see nothing on any screen, although I'm not sure where exactly it should appear.

**Comment by jgyates** [comment, marked as answer]
The error in your submitted logs shows your location ID is not valid:

Error in GetObservation: (627563311) : The searched item was not found. Reason: Unable to find the resource : myweather.py:120

Try using a 5 digit zip code if you are in the US. 

How did you determined your city ID?

**Reply by TEBrewMeister** [reply]
I followed the instructions on the Openweather site to create a GEOLocation Code, at least I think that's what they called it. I switched back to the zip code, was you suggested, saved it... and it appears to be working now. Thanks

**Reply by jgyates** [reply]
I am glad it is working now.. FYI, you typically do no create an ID, you use an existing one, This pages has some details:

https://www.dmopress.com/openweathermap-howto/

**Comment by zachliu** [comment]
You are not using the free tier right? I probably shouldn't be stingy on this :joy: 

```
Error in GetObservation: (94536) : Your API subscription level does not allow to perform this operation
Reason: Invalid API Key provided : myweather.py:118
```
